rpcserver.go

```diff
@@ -2776,43 +2779,41 @@ func createRPCOpenChannel(r *rpcServer, graph *channeldb.ChannelGraph,
 	}
 	externalCommitFee := dbChannel.Capacity - sumOutputs
 
-	chanID := dbChannel.ShortChannelID.ToUint64()
-
 	var (
 		uptime   time.Duration
 		lifespan time.Duration
 	)
 
 	// Get the lifespan observed by the channel event store.
-	startTime, endTime, err := r.server.chanEventStore.GetLifespan(chanID)
+	startTime, endTime, err := r.server.chanEventStore.GetLifespan(
+		dbChannel.FundingOutpoint,
+	)
 	if err != nil {
-		// If the channel cannot be found, log an error and do not perform
-		// further calculations for uptime and lifespan.
-		rpcsLog.Warnf("GetLifespan %v error: %v", chanID, err)
-	} else {
-		// If endTime is zero, the channel is still open, progress endTime to
-		// the present so we can calculate lifespan.
-		if endTime.IsZero() {
-			endTime = time.Now()
-		}
-		lifespan = endTime.Sub(startTime)
+		return nil, fmt.Errorf("GetLifespan %v error: %v",
```

Comment:
This is an independent change that could be an isolated commit. Not for me necessarily, but perhaps the second reviewer.